File: odds.py
import random, time
from itertools import product
from poker import *

start = time.clock()
init_rank()
end = time.clock()
print('Init took', '%.3f' % (end-start), 'seconds')

s, N, step = [0]*10, 0, 1000000
deck = list(v*16+s for v, s in product(range(2,15), range(4)))
while True:
    start = time.clock()
    for i in range(step): s[rank(random.sample(deck, 5))>>24] += 1
    end = time.clock()
    N += step
    print('%d:'%N, '  '.join('%.4f' % (100*v/N) for v in s))
    print('%.3f' % (end-start), 'seconds')

File: odds.cc
#include <vector>
#include <cstdlib>
#include <ctime>

using namespace std;

#include "poker.h"

int main() {
    clock_t start, end;

    start = clock();
    init_rank();
    end = clock();

    cout << "Init took " << (double)(end-start)/CLOCKS_PER_SEC << " seconds" << endl;

    vector<int> deck = make_deck();

    vector<int> cnt(10);
    start = clock();
    int N = 1000000;
    for(int i = 0; i < N; i++) {
        shuffle_deck(deck, 5);
        cnt[rank_hand(deck)>>24]++;
    }
    end = clock();
    cout << (double)(end-start)/CLOCKS_PER_SEC << " seconds" << endl;

    for(int i = 0; i < 10; i++) {
        cout << 100.0*cnt[i]/N << endl;
    }

    return 0;
}
